### PR TITLE
Pieti kinnunen/nav update

### PR DESCRIPTION
**Pull request by PietiKinnunen**
[Preview](https://docs-git-pietikinnunen-nav-update-goteleport.vercel.app/docs/)

- Update navigation structure
  - Add "Community" menu item
  - Move "Sign In" to the main nav
  - Replace previous "Sign In" button with "Contact Sales" button

Related PRs in other repos: [next](https://github.com/gravitational/next/pull/2200), [blog](https://github.com/gravitational/blog/pull/456)

---
Asana task: https://app.asana.com/0/1202613463734613/1205631765377005/f

### CODE REVIEW

File: components/Menu/Menu.tsx [hunk 1]
```diff
@@ -16,6 +35,38 @@ const Menu = () => {
...
+              href="https://teleport.sh/"
+              icon="clouds"
+              title="Teleport Cloud Login"
+              description="Login to your Teleport Account"
+            />
+            <DropdownMenuItemCTA
+              href="https://dashboard.gravitational.com/web/login/"
+              icon="download"
+              title="Dashboard Login"
+              description="Legacy Login &amp; Teleport Enterprise Downloads"
```
[the remark this answers is not in the document]
Comment (avatus) [reply components/Menu/Menu.tsx:65]: We recently removed the enterprise download link. Cloud/OSS still have their own download pages but enterprise is directed [here](https://github.com/gravitational/teleport/pull/31309/files#diff-fc5694a191191a903342c9b735e577af1485f86686d86400c217898362db4ff7R189). Not sure what would make the most sense here

[the remark this answers is not in the document]
Comment (PietiKinnunen) [reply components/Menu/Menu.tsx:65]: Seems that @travelton added changes to the next repo PR [a5733ad](https://github.com/gravitational/next/pull/2200/commits/a5733adbb1c6107b7cc9debedd2cbc48f99e283c) that remove the sign in dropdown and the link to the deprecated dashboard. I'll add those changes here as well.

[the remark this answers is not in the document]
Comment (PietiKinnunen) [reply components/Menu/Menu.tsx]: Addressed in [6ec2897](https://github.com/gravitational/docs/pull/403/commits/6ec2897ca1c5113c884bb97c90539576cdd5bd45)

File: components/Menu/Menu.tsx [hunk 2]
```diff
@@ -16,6 +35,38 @@ const Menu = () => {
...
+            isSignInVisible && styles.visible
+          )}
+          data-testid="sign-in-menu"
+        >
+          <DropdownMenuCTA title="Sign in to Teleport">
+            <DropdownMenuItemCTA
+              href="https://teleport.sh/"
+              icon="clouds"
+              title="Teleport Cloud Login"
+              description="Login to your Teleport Account"
```
[the remark this answers is not in the document]
Comment (PietiKinnunen) [reply components/Menu/Menu.tsx]: The dropdown was removed in [6ec2897](https://github.com/gravitational/docs/pull/403/commits/6ec2897ca1c5113c884bb97c90539576cdd5bd45)

**Review by travelton — APPROVED** [review]
Good to go from me.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**